docs/ota-client-guide/modules/ROOT/pages/garage-sign.adoc

```diff
@@ -0,0 +1,256 @@
+
+* xref:rotating-signing-keys.adoc#_rotate_the_online_keys_with_your_new_offline_keys[want to rotate your online keys with your new offline keys].
+* believe your keys have been compromised.
+* would like to revoke the current keys for signing metadata.
+
+==== Workflow to generate new keys
+
+When generating new keys for metadata, you perform the following steps:
+
+. First, generate the new `root` and `targets` keys by running the command `garage-sign key generate` twice.
++
+[source,bash]
+----
+garage-sign key generate \
+  --repo <reponame> \
+  --name myroot \ <1>
+  --type <ed25519|rsa> <2>
+garage-sign key generate \
+  --repo <reponame> \
+  --name mytargets \
+  --type <ed25519|rsa>
+----
++
+<1> You can give any name to the keys you generate, which helps to easily identify between the new `root` and `targets` keys. Here, the keys generated are given the names `myroot` and `mytargets`.
+<2> The key type must be specified as either Ed25519 or RSA. In OTA Connect, an RSA key is used for signing uptane metadata.
+
+. Afterwards, pull the latest `targets.json` file from OTA Connect with the command `garage-sign targets pull`.
++
+[source,bash]
+----
+garage-sign targets pull --repo <reponame>
+----
+
+. Next, use the command `garage-sign move-offline`. This command deletes the old `root` key from OTA Connect, downloads and signs the new `root` key with the newly generated keys, and pushes the new `root.json` file to OTA Connect.
++
+[source,bash]
+----
+garage-sign move-offline \
+  --repo <reponame> \
+  --old-root-alias originroot \ <1>
+  --new-root myroot \ <2>
+  --new-targets mytargets <3>
+----
++
+<1> This is an example of an alias given to the old `root` key. It gets saved with this name.
+<2> This is an example of the name given to your new `root` key which you previously generated using the command `garage-sign key generate`.
+<3> This is an example of the name given to your new `targets` key which you previously generated using the command `garage-sign key generate`.
+
+. As a final step, sign the new `targets.json` with your newly generated `targets` key and upload it to OTA Connect.
++
+[source,bash]
+----
+garage-sign targets sign --repo <reponame> --key-name mytargets
+garage-sign targets push --repo <reponame>
+----
+
+Please refer to xref:rotating-signing-keys.adoc#_rotate_the_keys_for_root_and_targets_metadata[rotating the keys for `root` and `targets` metadata] for additional information on generating new keys.
+
+=== Add a new software version
+
+`garage-sign` can also be used to add software packages to your repository with the command `garage-sign targets add`. The packages are downloaded by aktualizr, however `garage-sign` doesn't upload the software file. The `garage-sign targets add` commmand puts new metadata in place, hence you need to tell aktualizr where the software package can be found. This is done by uploading the file somewhere else and providing `garage-sign` with the URL of where it's available.

```

Comment:
This is all over the place. You use "software packages" and "software file" without any clear idea of what the difference is, and you don't use the actual term we use in OTA Connect. Please go back to the drawing board.